inc/gpio.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+// GPIO HAL Interface
+
+#include <stdbool.h>
+#include <stdint.h>
+
+// IO Pin "address" to be used to change that pin's settings
+// Use bytes instead of words to save on memory also convenient for MSP430
+// compatibility
+typedef struct IOAddr {
+  const uint8_t IOPort;  // Could be an enum (A-F) for stm32f0xx
+  const uint8_t IOPin;
+} IOAddr;
```

Comment:
I attempted to fix this but it may need more discussion, see what you think in new commit.